payjoin/src/send/mod.rs

```diff
@@ -486,10 +494,12 @@ impl Serialize for RequestContext {
         let mut state = serializer.serialize_struct("RequestContext", 8)?;
         state.serialize_field("psbt", &self.psbt.to_string())?;
         state.serialize_field("endpoint", &self.endpoint.as_str())?;
-        let ohttp_string = self
-            .ohttp_config
-            .as_ref()
-            .map_or("".to_string(), |config| bitcoin::base64::encode(config.encode().unwrap()));
+        let ohttp_string = self.ohttp_config.as_ref().map_or(Ok("".to_string()), |config| {
+            config
+                .encode()
+                .map_err(|e| serde::ser::Error::custom(format!("ohttp-keys encoding error: {}", e)))
```

Comment:
whats the motivation for using `Error::custom` here and not doing ` .map_err(|e| InternalCreateRequestError::V2(e.into()))?` as above?

Reply:
We have to return `Result<Serializer::Ok, Serializer::Error>` here where CreateRequestError is not Serializer::Error. Above is in a function that returns `Result<(Request, ContextV2), CreateRequestError>` where that would be InternalCreateRequestError has a conversion into CreateRequestError.